bin/ebs-autoscale

```diff
@@ -178,34 +142,34 @@ LOG_COUNT=$LOG_INTERVAL
 # time in seconds between event loops
 # keep this low so that rapid increases in utilization are detected
 DETECTION_INTERVAL=$(get_config_value .detection_interval)
-# get the number of devices once when the script first starts
-NUM_DEVICES=$(get_num_devices)
-THRESHOLD=$(calc_threshold "${NUM_DEVICES}")
 
-while true; do
+function refresh() {
+  NUM_DEVICES=$(get_num_devices)
+  THRESHOLD=$(calc_threshold "${NUM_DEVICES}")
   STATS=$(df -BG  ${MOUNTPOINT} | grep -v Filesystem)
   TOTAL_SIZE=$(echo ${STATS} | awk '{print $2}')
   USED=$(echo ${STATS} | awk '{print $3}')
   AVAILABLE=$(echo ${STATS} | awk '{print $4}')
   PCT_UTILIZATION=$(echo ${STATS} | awk '{print $5}' | cut -d"%" -f1 -)
+}
+
+while true; do
+  refresh
   if  [ $PCT_UTILIZATION -ge "${THRESHOLD}" ]; then
-    # get number of devices only when we need to add more space
-  	NUM_DEVICES=$(get_num_devices)
```

Comment:
I am not sure why this logic is removed.

Reply:
same comment:  it's moved into `refresh` function. it's no longer an aws call, just a read from the file so it can happen on every loop.